plugins/inputs/influxdb_listener/influxdb_listener_test.go

```diff
@@ -155,6 +155,68 @@ func TestWriteBasicAuth(t *testing.T) {
 	require.EqualValues(t, http.StatusNoContent, resp.StatusCode)
 }
 
+func TestWriteHTTPTransformHeaderValuesToTags(t *testing.T) {
+	listener := newTestListener()
+	listener.HTTPHeaderTags = map[string]string{"Present_http_header_1": "presentMeasurementKey1", "Present_http_header_2": "presentMeasurementKey2", "NOT_PRESENT_HEADER": "notPresentMeasurementKey"}
+
+	acc := &testutil.Accumulator{}
+	require.NoError(t, listener.Init())
+	require.NoError(t, listener.Start(acc))
+	defer listener.Stop()
+
+	// post single message to listener
+	req, err := http.NewRequest("POST", createURL(listener, "http", "/write", "db=mydb"), bytes.NewBuffer([]byte(testMsg)))
+	require.NoError(t, err)
+	req.Header.Set("Content-Type", "")
+	req.Header.Set("Present_http_header_1", "PRESENT_HTTP_VALUE_1")
+	req.Header.Set("Present_http_header_2", "PRESENT_HTTP_VALUE_2")
+	resp, err := http.DefaultClient.Do(req)
+	require.NoError(t, err)
+	resp.Body.Close()
+	require.EqualValues(t, 204, resp.StatusCode)
+
+	acc.Wait(1)
+	acc.AssertContainsTaggedFields(t, "cpu_load_short",
+		map[string]interface{}{"value": float64(12)},
+		map[string]string{"host": "server01", "presentMeasurementKey1": "PRESENT_HTTP_VALUE_1", "presentMeasurementKey2": "PRESENT_HTTP_VALUE_2"},
+	)
+
+	// post single message to listener with a database tag in it already. It should be clobbered.
```

Comment:
Could you split each test case into a separate tests or use a table test?

https://github.com/golang/go/wiki/TableDrivenTests
https://github.com/influxdata/telegraf/blob/7b03a6f4abbf96cbb0c5fc7497071b34c19c9ee8/plugins/inputs/influxdb_listener/influxdb_listener_test.go#L549-L589

Reply:
I split the single write and bulk write into separate tests, I think that makes more sense here than using a table test